iep-007/README.adoc

```diff
@@ -0,0 +1,153 @@
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+endif::[]
+
+= IEP-007: Kubernetes cluster upgrade
+
+:toc:
+
+.Metadata
+[cols="2"]
+|===
+| IEP
+| 007
+
+| Title
+| Kubernetes cluster upgrade 
+
+| Author
+| link:https://github.com/olblak[Olblak]
+
+| Status
+| :speech_balloon: In-process
+
+| Type
+| [Process]
+
+| Created
+| 2016-07-04
+|===
+
+
+
+== Abstract
+
+As Kubernetes project often releases new versions, it's interesting to enjoy new features or bugfixes introduced by those new versions. 
+Therefor we need an easy way to upgrade Kubernetes cluster on a regular basis.
+
+
+== Specification
+
+Currently there is two main strategies to upgrade a cluster. +
+
+. Upgrading an existing cluster.
+. Migration on a second cluster.
+
+As Azure do not (yet) provide any tools to upgrade existing clusters, we have to upgrade them manually.  +
+It appears to be easier and safer to deploy a new cluster and re-deploy all resources on the new one.
+
+Until the new cluster stays in the same region than the previous one, we can keep persistent data from blob storage and attach them to the new cluster. +
+The only important element that we loose when we migrate to a new cluster, is the cluster public IP. +
+This means that we need to update 'nginx.azure.jenkins.io' with the public IP.
+
+=== Migration Process
+
+IMPORTANT: The old cluster must be kept until the new one is ready to served requests.
+
+==== Step 1 : Backup
+
+. Backup secrets containing Letsencrypt certificates. (Manual operation)
+----
+mkdir ~/tls
+.bin/kubectl get secret accountapp-tls --export=true  -o yaml > ~/tls/accountapp.yaml
+.bin/kubectl get secret pluginsite-tls --export=true  -o yaml > ~/tls/pluginsite.yaml
+.bin/kubectl get secret repo-proxy-tls --export=true  -o yaml > ~/tls/repo-proxy.yaml
+----
+
+==== Step 2 : Deploy the new cluster
+
+. Add a second k8s resource in github.com/jenkins-infra/azure, named 'peak8s' (Require PR on jenkins-infra/azure)
+. Update following hieraconfig variables with new the k8s cluster informations(Require PR on jenkins-infra/jenkins-infra)
+----
+profile::kubernetes::kubectl::server 
```

Comment:
The way this Puppet operation is defined right now means that when we decide to deploy the new cluster, we will lose the ability to do any management of the old cluster.

I'm wondering if we should make the parameters (server, username, etc) an array of k8s clusters and just teach Puppet how to manage many clusters at once.